service/frontend/fx.go

```diff
@@ -362,27 +362,39 @@ func NamespaceRateLimitInterceptorProvider(
 		panic("invalid service name")
 	}
 
-	rateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
+	namespaceRateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
 		MemberCounter:    frontendServiceResolver,
 		PerInstanceQuota: serviceConfig.MaxNamespaceRPSPerInstance,
 		GlobalQuota:      globalNamespaceRPS,
 	}.GetQuota
+	namespaceBurstFn := func(ns string) int {
+		return int(namespaceRateFn(ns) *
+			serviceConfig.MaxNamespaceBurstRatioPerInstance(ns))
```

Comment:
Need to make sure this is equal or greater than 1. And echo what Yichao suggested, move this into the NewNamespaceRateBurst

Reply:
@yiminc, how about we always take BurstRatio = min(1, BurstRatio). This way burst is atleast RPS. Reducing burst below 1 will reduce rps. If someone wants to reduce RPS they should not use burst config for that. This will prevent burst from accidently reducing RPS.